pennylane/ops/arithmetic/sum.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Sum class which contains logic for
+computing the sum of operations.
+"""
+import warnings
+import numpy as np
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import Operator, expand_matrix
+
+
+def sum(*summands, wire_order=None):
+    """Compute the sum of the provided terms"""
+    return Sum(*summands, wire_order=wire_order)  # a wire order is required when combining operators of varying sizes
+
+
+class Sum(Operator):
+    """Arithmetic operator subclass representing the sum of operators"""
+
+    def __init__(self, *summands, do_queue=True, id=None):
+
+        if len(summands) < 2:
+            raise ValueError(f"Require at least two operators to sum; got {len(summands)}")
+
+        self.summands = summands
```

Comment:
Should we also stick these in the hyperparameters?
`self.hyperparameters['summands'] = summands`

Reply:
As mentioned before, I am not sure about this. Technically, it obviously defines and changes the operator, so it should be a hyperparam. But it is such a big one, maybe this warrants a special attribute like done here.

I prefer the latter.

Reply:
I agree, especially since the summands define the operation, kind of like how the angle parameter defines the `Rx` rotation operator